Compiler/NFFrontEnd/NFClass.mo

```diff
@@ -136,15 +136,63 @@ uniontype Class
     input array<InstNode> components;
     input Class expandedClass;
     output Class instancedClass;
+  protected
+    list<Equation> eqs = {}, eqs1 = {}, eqs2 = {};
+    list<Equation> ieqs = {}, ieqs1 = {}, ieqs2 = {};
+    list<list<Statement>> algs = {}, algs1 = {}, algs2 = {};
+    list<list<Statement>> ialgs = {}, ialgs1 = {}, ialgs2 = {};
   algorithm
     instancedClass := match expandedClass
       case EXPANDED_CLASS()
-        then INSTANCED_CLASS(expandedClass.elements, components,
-          expandedClass.equations, expandedClass.initialEquations,
-          expandedClass.algorithms, expandedClass.initialAlgorithms);
+        algorithm
+          eqs1 := expandedClass.equations;
+          ieqs1 := expandedClass.initialEquations;
+          algs1 := expandedClass.algorithms;
+          ialgs1 := expandedClass.initialAlgorithms;
+
+          // append the sections from the inherited classes
+          for ext in expandedClass.extendsNodes loop
+            (eqs2, ieqs2, algs2, ialgs2) := collectInherited(ext, eqs2, ieqs2, algs2, ialgs2);
+          end for;
+          eqs := listAppend(eqs2, eqs1);
+          ieqs := listAppend(ieqs2, ieqs1);
+          algs := listAppend(algs2, algs1);
+          ialgs := listAppend(ialgs2, ialgs1);
```

Comment:
If i don't do this the DAE gets no equations from the inherited classes. Maybe it should be done sompeplace else but it needs to be done.